src/DGmethods/test/Sphere/advection_sphere.jl

```diff
@@ -0,0 +1,290 @@
+# Advection on the Sphere test case.  This is a case similar to the one presented in:
+#
+#@article{WILLIAMSON1992211,
+#title = "A standard test set for numerical approximations to the shallow water equations in spherical geometry",
+#journal = "Journal of Computational Physics",
+#volume = "102",
+#number = "1",
+#pages = "211 - 224",
+#year = "1992",
+#issn = "0021-9991",
+#doi = "https://doi.org/10.1016/S0021-9991(05)80016-6",
+#url = "http://www.sciencedirect.com/science/article/pii/S0021999105800166",
+# }
+#
+# This version runs the advection on the sphere as a stand alone test (no dependence
+# on CLIMA moist thermodynamics)
+
+using MPI
+using CLIMA.Topologies
+using CLIMA.Grids
+using CLIMA.DGBalanceLawDiscretizations
+using CLIMA.DGBalanceLawDiscretizations.NumericalFluxes
+using CLIMA.MPIStateArrays
+using CLIMA.LowStorageRungeKuttaMethod
+using CLIMA.ODESolvers
+using CLIMA.GenericCallbacks
+using LinearAlgebra
+using StaticArrays
+using Logging, Printf, Dates
+
+const uid, vid, wid = 1:3
+const integration_testing = true
+const γ_exact = 7 // 5
+
+if !@isdefined integration_testing
+  const integration_testing =
+    parse(Bool, lowercase(get(ENV,"JULIA_CLIMA_INTEGRATION_TESTING","false")))
+  using Random
+end
+
+# preflux computation: NOT needed for this test
+@inline function preflux(Q, _...)
+  γ::eltype(Q) = γ_exact
+  @inbounds ρ = Q[1]
+    ρinv = 1 / ρ
+    u, v, w, P = 0, 0, 0, 0
+    (P, u, v, w, ρinv)
+end
+
+#{{{ advectionflux
+advectionflux!(F, Q, QV, aux, t) = advectionflux!(F, Q, QV, aux, t, preflux(Q)...)
+@inline function advectionflux!(F,Q,QV,aux,t,P,u,v,w,ρinv)
+    @inbounds begin
+        u,v,w = aux[uid], aux[vid], aux[wid]
+        ρ=Q[1]
+        F[1,1], F[2,1], F[3,1]=u*ρ, v*ρ, w*ρ
+    end
+end
+#}}} advectionflux
+
+#{{{ wavespeed
+@inline function wavespeed(n, Q, vel, t...)
+    abs(dot(vel,n))
+end
+#}}} wavespeed
+
+#{{{ Cartesian->Spherical
+@inline function cartesian_to_spherical(x,y,z,format)
+    #Conversion Constants
+    if format == "degrees"
+        c=180/π
+    elseif format == "radians"
+        c=1.0
+    end
+    λ_max=2π*c
+    λ_min=0*c
+    ϕ_max=+0.5*π*c
+    ϕ_min=-0.5*π*c
+
+    #Conversion functions
+    r = hypot(x,y,z)
+    λ=atan(y,x)*c
+    ϕ=asin(z/r)*c
+    return (r, λ, ϕ)
+end
+#}}} Cartesian->Spherical
+
+#{{{ velocity initial condition
+@inline function velocity_init!(vel, x, y, z)
+    @inbounds begin
+        DFloat = eltype(vel)
+        (r, λ, ϕ) = cartesian_to_spherical(x,y,z,"radians")
+        #w = 2 * DFloat(π) * cos(ϕ) #Case 1 -> shear flow
+        w = 2 * DFloat(π) * cos(ϕ) * r #Case 2 -> solid body flow
+        uλ, uϕ = w, 0
+        vel[uid] = -uλ*sin(λ) - uϕ*cos(λ)*sin(ϕ)
+        vel[vid] = +uλ*cos(λ) - uϕ*sin(λ)*sin(ϕ)
+        vel[wid] = +uϕ*cos(ϕ)
+    end
+end
+#}}} velocity initial condition
+
+# initial condition
+function advection_sphere!(Q, t, x, y, z, vel)
```

Comment:
`t` is not used, so I'd recommend removing it (since a user might think they can run the test shorter, but really you have to do a full rotation for the error check to work.